When is the PTO armed from? (#3959)
quicwg/base-drafts: draft-ietf-quic-recovery.md

A sender recomputes and may need to reset its PTO timer every time an
ack-eliciting packet is sent or acknowledged, when the handshake is confirmed,
or when Initial or Handshake keys are discarded. This ensures the PTO is always
set based on the latest RTT information and for the last sent ack-eliciting
packet in the correct packet number space. When the PTO expires and there are

> no ack-eliciting packets in flight, the PTO is set from the current time.

Reviewer: ```suggestion
no ack-eliciting packets in flight, the PTO is set from that moment.
```

keys indicates forward progress and the loss detection timer might have been set
for a now discarded packet number space.

> #### Before Server Address Validation

Reviewer 2: Client?

Author: It's before the Client is sure the Server has validated the client's address, but I'll remove this change since it's unrelated.

until it is certain that the server has finished its address validation
(see Section 8 of {{QUIC-TRANSPORT}}).  That is, the client MUST set the
probe timer if the client has not received an acknowledgement for one of its

> Handshake or 1-RTT packets, and has not received a HANDSHAKE_DONE frame,

Reviewer: ```suggestion
Handshake or 1-RTT packets and has not received a HANDSHAKE_DONE frame,
```

> even if there are no packets in flight. If Handshake keys are available to the
> client, it MUST send a Handshake packet, and otherwise it MUST send an Initial
> packet in a UDP datagram of at least 1200 bytes.

Reviewer: ```suggestion
even if there are no packets in flight. When the PTO fires, the client MUST
send a Handshake packet if it has Handshake keys, otherwise it MUST
send an Initial packet in a UDP datagram of at least 1200 bytes.
```
